packages/text-input/src/TextInput.tsx

```diff
@@ -278,7 +294,7 @@ const TextInput = React.forwardRef(
             />
           )}
 
-          {state === State.Error && (
+          {state === State.Error && errorMessage && (
```

Comment:
So this I think should not depend on the existence of `errorMessage`, only for showing the message.

Such that someone could show an error state but not show a message if they desire.